src/backend/distributed/utils/metadata_cache.c

```diff
@@ -232,6 +232,21 @@ LoadShardInterval(uint64 shardId)
 
 
 /*
+ + * ReplicationModel returns the replication model for the given relation. The function
+ + * errors out for tables that are not distributed.
+ + */
+char
+ReplicationModel(Oid relationId)
```

Comment:
I'm *VERY* doubtful that these kind of functions are a good idea. There's already functions that do
```
Var *partitionKey = PartitionKey();
char partitionMethod = PartitionMethod();
```

which is quite wasteful, because it does repeated cache-lookups.  Adding yet another function that'll be just added alongside seems wrong. We should really just use the cache entry directly.